src/Streamly/Internal/Data/Parser.hs

```diff
@@ -668,23 +672,39 @@ escapedFrameBy _begin _end _escape _p = undefined
 -- separated text.
 --
 -- * Stops - when it finds a word separator after a non-word element
+-- | Ignore every element on which the predicate succeeds before
+-- an element where predicate fails is found, then ignore every
+-- element after this which fails the predicate until another
+-- element which passes the predicate is found, put this second
+-- succeeding element back into the input.
+--
+-- Examples: -
+--
+-- parseWord = S.parse (PR.wordBy (==\'.\') FL.toList)
+--
+-- 1. parseWord ...a...b...
+-- gives [a], while b... is unread
+-- 2. parseWord ....b....
+-- gives [b], while nothing is unread
+-- 3. parseWord .....
+-- gives [], while nothing is unread
+--
+-- * Stops - when it finds a non-word after a non-word element
```

Comment:
`Stops - when it finds a word separator after a non-word element`?
Is a non-word element a word separator?
Maybe you meant, `Stops - when it finds a word after a word separator`?